### DISCUSSION TITLE
arco design pro （vue） 修改组件的颜色无效怎么办

### DISCUSSION THREAD

**Discussion by aloniacode** [arco-design/arco-design-vue#1806] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/55039022/198860484-1ad135db-5549-4e5c-bb67-b811d5069b4a.png)
没用啊

**Comment by hehehai** [comment, marked as answer]
@coderlcb hi, 可以参考该 [问题](https://github.com/arco-design/arco-design-vue/issues/1503#issuecomment-1216364074)

参考：https://stackblitz.com/edit/vitejs-vite-qy4cey?file=vite.config.ts （使用了 vite3）

```ts
import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import Components from 'unplugin-vue-components/vite';
import { ArcoResolver } from 'unplugin-vue-components/resolvers';

// https://vitejs.dev/config/
export default defineConfig({
  css: {
    preprocessorOptions: {
      less: {
        modifyVars: { 'color-primary-6': '#13c2c2' },
      },
    },
  },
  plugins: [
    vue(),
    Components({
      resolvers: [
        ArcoResolver({
          importStyle: 'less',
        }),
      ],
    }),
  ],
});
```

<img width="400" alt="image" src="https://user-images.githubusercontent.com/12692552/184843618-0fc89913-d5ca-48a2-ad29-88f4d6f60385.png">

less 变量可以使用 css 变量来调整（明暗模式影响的是 css 变量）

```ts
css: {
    preprocessorOptions: {
      less: {
        modifyVars: {
          'primary-6': `rgb(var(~'@{arco-cssvars-prefix}-green-6'))`,
          'dark-primary-6': `rgb(var(~'@{arco-cssvars-prefix}-orange-6'))`,
        },
      },
    },
  },
```

这里主题自定义建议使用官方的[主题设计系统来配置](https://arco.design/themes)

**Reply by aloniacode** [reply]
没有用。。。。

**Reply by hehehai** [reply]
用 Stackblitz 发一个复现吧

**Reply by aloniacode** [reply]
> 用 Stackblitz 发一个复现吧

找到问题了,pro项目初始化后main.ts默认导入了组件库的css文件，换成less就可以了。
